lib/Basic/LangOptions.cpp

```diff
@@ -230,6 +230,8 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
     addPlatformConditionValue(PlatformConditionKind::OS, "PS4");
   else if (Target.isOSHaiku())
     addPlatformConditionValue(PlatformConditionKind::OS, "Haiku");
+  else if (Target.isOSBinFormatWasm())
```

Comment:
I'm not sure that we want to introduce this at all.  There shouldn't be anything specific to WASM here, it is an object file format, not an OS.  We should be introducing a `none` OS (which is the canonical spelling for this field being unset).

Reply:
Hmm, AFAIK `llvm::Triple` can't represent `none` OS you said.
You mean we should add `none` case to `llvm::Triple`?

Reply:
> There shouldn't be anything specific to WASM here, it is an object file format, not an OS. 

I agree that and I think it's necessary to handle `isOSWASI` case. 

>  We should be introducing a none OS (which is the canonical spelling for this field being unset).

You mean we should add `none` enum case to `llvm::Triple`? What the difference of `UnknownOS`?

Reply:
`none` is the canonical spelling for *no* OS, `unknown` is meant for the vendor field only.  There is no such thing as an unknown OS, you either have an OS or do not.  The compiler needs to know what the code is being generated for.

Reply:
OK, I understood what `none` OS means. Again, you mean we should add `llvm::Triple::None`, right?